src/Services/SharesHandler.php

```diff
@@ -19,40 +19,40 @@ public function __construct(EntityManagerInterface $entityManager)
     }
 
     /**
-     * @param SharePostRequest $request
+     * @param ShareItemRequest $request
      *
      * @throws \Exception
      */
-    public function sharePost(SharePostRequest $request)
+    public function shareItem(ShareItemRequest $request)
     {
-        $request->getPost()->getSharedPosts()->clear();
+        $request->getItem()->getSharedItems()->clear();
 
         foreach ($request->getUsers() as $user) {
             $inbox = $user->getInbox();
 
-            $post = new Post();
-            $post->setParentList($inbox);
-            $post->setSecret($request->getPost()->getSecret()); // TODO логика и шифрование на фронтовой стороне
-            $post->setOriginalPost($request->getPost());
+            $item = new Item();
+            $item->setParentList($inbox);
+            $item->setSecret($request->getItem()->getSecret()); // TODO логика и шифрование на фронтовой стороне
+            $item->setOriginalItem($request->getItem());
 
-            $this->entityManager->persist($post);
+            $this->entityManager->persist($item);
         }
 
         $this->entityManager->flush();
     }
 
     /**
-     * @param Post $post
+     * @param Item $item
      *
      * @throws \Exception
      */
-    public function savePostWithShares(Post $post)
+    public function saveItemWithShares(Item $item)
     {
-        $this->entityManager->persist($post);
-        foreach ($post->getSharedPosts() as $sharedPost) {
-            $sharedPost->setSecret($post->getSecret());
+        $this->entityManager->persist($item);
+        foreach ($item->getSharedItems() as $sharedItem) {
+            $sharedItem->setSecret($item->getSecret());
```

Comment:
You just copy secret string for share? It need to be encrypted with public user key on frontend. Isn't it?